lib/metasploit/credential/importer/core.rb

```diff
@@ -140,57 +134,105 @@ def import_long_form
+         public: public_object,
+         private: private_object_for_row,
+         realm: realm_object_for_row}
+
+        rows << row
+
+
+
+      end
+      if all_creds_valid
+        core_opts.each_index do |index|
+          row = rows[index]
+
+
+          # Host and Service information for Logins
+          host_address      = row['host_address']
+          service_port      = row['service_port']
+          service_protocol  = row['service_protocol']
+          service_name      = row['service_name']
+          # These were not initially included in the export, so handle
+          # legacy cases:
+          access_level      = row['access_level'].present? ? row['access_level'] : ''
+          last_attempted_at = row['last_attempted_at'].present? ? row['last_attempted_at'] : ''
+          status            = row['status'].present? ? row['status'] : ''
+
+          if Metasploit::Credential::Core.where(core_opts[index]).blank?
+            core = create_credential_core(core_opts[index])
+          else
+            core = Metasploit::Credential::Core.where(core_opts[index])
+          end
+
+
+          if host_address.present? && service_port.present? && service_protocol.present?
+            login_opts = {
+                core: core,
+                address: host_address,
+                port: service_port,
+                protocol: service_protocol,
+                workspace_id: workspace.id,
+                service_name: service_name.present? ? service_name : ""
+            }
+            login_opts[:last_attempted_at] = last_attempted_at unless status.blank?
+            login_opts[:status]            = status unless status.blank?
+            login_opts[:access_level]      = access_level unless access_level.blank?
 
-        core = create_credential_core(origin:origin, workspace_id: workspace.id,
-                                                     public: public_object,
-                                                     private: private_object_for_row,
-                                                     realm: realm_object_for_row )
-
-        # Make Logins with attendant Host/Service information if we are doing that
-        if host_address.present? && service_port.present? && service_protocol.present?
-          login_opts = {
-            core: core,
-            address: host_address,
-            port: service_port,
-            protocol: service_protocol,
-            workspace_id: workspace.id,
-            service_name: service_name.present? ? service_name : ""
-          }
-          login_opts[:last_attempted_at] = last_attempted_at unless status.blank?
-          login_opts[:status]            = status unless status.blank?
-          login_opts[:access_level]      = access_level unless access_level.blank?
-
-          create_credential_login(login_opts)
+            create_credential_login(login_opts)
+
+          end
         end
       end
-    end
+      end
+    return all_creds_valid
```

Comment:
No need to return this. We can call valid? on the instance of this object and have all the validators run. It would achieve the same thing.